src/_posts/databases/mysql/2000-01-01-troubleshooting.md

```diff
@@ -0,0 +1,169 @@
+---
+title: Troubleshooting Scalingo for MySQL®
+nav: Troubleshooting
+modified_at: 2024-03-12 12:00:00
+tags: databases mysql addon
+index: 10
+---
+
+## Understanding Memory Consumption
+
+MySQL® uses several strategies to efficiently manage data and optimize
+performances. Among them, MySQL® (and more specifically InnoDB) maintains a
+storage area called *Buffer Pool* for caching data, indexes, and other
+auxiliary buffers in memory. This cache helps reduce disk I/O, and, thus, speed
+up query execution.
+
+Scalingo for MySQL® instances typically allocate:
+- ~50% of the memory to the [Buffer Pool](https://dev.mysql.com/doc/refman/8.0/en/innodb-buffer-pool-resize.html)
+- ~37.5% to [Connection Management](https://dev.mysql.com/doc/refman/8.0/en/connection-management.html)
+- ~12.5% to [Group Replication](https://dev.mysql.com/doc/refman/8.0/en/group-replication.html)
+
+MySQL® generally tries to avoid swapping to disk because disk I/O is
+significantly slower than memory access. However, if the system is under
+memory pressure, the underlying operating system can decide to swap out MySQL®
+memory pages to disk, leading to performance degradation and increased latency.
+
+As a consequence, the size of the Buffer Pool should ideally be large enough to
+handle the database workload, thus preventing swap usage. A constantly swapping
+database could indicate that the space dedicated to the Buffer Pool has become
+insufficient. In such a case, [switching to a superior plan]({% post_url databases/mysql/2000-01-01-managing %}#scaling-changing-plan)
+should quickly resolve the swapping issue. Beware though that a Buffer Pool
+that is too large can also cause swapping, due to competition for memory.
```

Comment:
> Beware though that a Buffer Pool that is too large can also cause swapping, due to competition for memory.


question: I don't really understand this sentence.

Reply:
If the buffer pool is too large, the OS might have to swap for other processes to run properly, which would also be bad.
Maybe I could rephrase this a bit, you're right :)

Reply:
I just removed the sentence, it was a bit out of scope.